atlasdb-cassandra/src/main/java/com/palantir/atlasdb/keyvalue/cassandra/CassandraKeyValueService.java

```diff
@@ -2123,6 +2127,53 @@ public void compactInternally(TableReference tableRef) {
         }
     }
 
+    @Override
+    public NodeAvailabilityStatus getNodeAvailabilityStatus() {
+        try {
```

Comment:
This method of getting availability status works in the standard configuration (3 nodes, RF3, or more generally numNodes = RF). However, if there are more nodes than the RF, then we're only checking whether we can do these reads for the `_locks` table - we could get lucky and return `EACH_QUORUM_AVAILABLE` when only 2 out of 6 nodes are up.

In the general case, it might be better to determine exactly how many nodes are up/down, using a similar healthcheck as is in CassandraClientPool.